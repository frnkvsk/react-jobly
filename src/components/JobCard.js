import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
    minWidth: 275,
    margin: '10px 10px 0 10px',
    boxShadow: '0 2px 4px rgba(0,0,0,0.3)',
    "&:hover": {
      boxShadow: '0 1px 2px rgba(0,0,0,0.3)'
    },
    cursor: 'pointer',
    backgroundColor: '#ffffff',
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    margin: '10px 0 10px',
  },
  pos: {
    margin: '0 0 0',
  },
});

export default function JobCard({job}) {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardContent>        
        <Typography className={classes.title} variant="h5" component="h2">
          {job.title}
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          Salary: {job.salary}
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          Equity: {job.equity}
        </Typography>        
      </CardContent>
      <CardActions>
      <Button className={classes.btn} variant="contained" color="secondary">
        Apply
      </Button>
      </CardActions>
    </Card>
  );
}

