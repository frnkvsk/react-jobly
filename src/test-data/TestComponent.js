// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

