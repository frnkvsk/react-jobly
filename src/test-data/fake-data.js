const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }