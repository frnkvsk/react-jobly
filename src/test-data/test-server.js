import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
