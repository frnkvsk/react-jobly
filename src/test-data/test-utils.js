// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
