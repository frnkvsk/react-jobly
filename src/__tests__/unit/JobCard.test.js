import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../test-data/test-utils';
import JobCard from '../../components/JobCard';
import { testData } from '../../test-data/fake-data';


// smoke and appearance tests
test('renders JobCard ', async() => {
  render(<JobCard job={testData.jobs[0]}/>);
  
  expect(screen.getByText(/Editor, magazine features/i)).toBeInTheDocument();

  expect(screen.getByText(/0.115/i)).toBeInTheDocument();  
  
});


// snapshot test
// test('matches snapshot', () => {
//   const { asFragment } = render(<JobCard job={testData.jobs[0]}/>);
//   expect(asFragment()).toMatchSnapshot();
// });