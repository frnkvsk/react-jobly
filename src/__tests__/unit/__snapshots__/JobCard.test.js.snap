// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root makeStyles-title-7 MuiTypography-h5"
      >
        Editor, magazine features
      </h2>
      <p
        class="MuiTypography-root makeStyles-pos-8 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Salary: 11111
      </p>
      <p
        class="MuiTypography-root makeStyles-pos-8 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Equity: 0.115
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Apply
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
